### DISCUSSION TITLE
[2023-09-25] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#7887] [Weekly Demo Day]
* Merged #7804
* The branch with fix for operand argument placeholder didn't work, and needed a few more days of fixing: #7530 is ready to merge
* Started #7785 The code is complete, but still miss some parts required to actually see it in action (it turned out the execution context has to be created to receive any update from the Language Server).

Here is presentation of argument placeholders:

https://github.com/enso-org/enso/assets/3919101/f7773412-5f0f-46c9-9848-2dc653aee882

**Comment by MichaelMauderer** [comment]
* Finished integration of GUI2 Code Editor (Code Mirror) [#7734](https://github.com/enso-org/enso/issues/7734)


[Peek 2023-09-25 12-12.webm](https://github.com/enso-org/enso/assets/1428930/58a926de-1042-4f2b-8e2c-29da953816b0)

* Working on expression updates https://github.com/enso-org/enso/issues/7783

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/7845 
- https://github.com/enso-org/enso/pull/7779 more issues fixed in the implementation, still on QA
- https://github.com/enso-org/enso/pull/7844 Work in progress, there is a huge issue with focus management.
- https://github.com/enso-org/enso/pull/7802 small contribution to the GUI2
- https://github.com/enso-org/enso/issues/7823 work in progress

<img width="836" alt="Screenshot 2023-09-25 at 5 57 49 PM" src="https://github.com/enso-org/enso/assets/6566674/5887bb38-0ffd-417f-82d3-6daa00039044">

**Comment by Frizi** [comment]
- Finished integration of gui2 and dashboard into single repo-wide npm workspace, which allowed the language server connection PR to pass CI. #7735
- Currently working on sending text updates to language server #7880

**Comment by somebody1234** [comment]
- Finished refactoring #7773 (almost ready to merge) (no new UI, video not added)
- Got this PR working: #7873

https://github.com/enso-org/enso/assets/4046547/4e743969-13c0-4288-ae58-c41b976076b0

- Beginning #7874 and #7865:

https://github.com/enso-org/enso/assets/4046547/cb1fd963-41ad-4115-8ebb-56f2ece8caa3

**Comment by wdanilo** [comment]
New website progress:

1. Finalized texts across the website.
2. I was working on the nodes animation with James and Sylwia – what to show, when, what to emphasize.
3. Added new section to the website. The only missing piece now is finalizing the animation of nodes on the website and small stuff, like adding pricing part. We should start going out with it next week.


https://github.com/enso-org/enso/assets/1623053/996de052-c963-4ec7-bdfc-f110d8655687

**Comment by mwu-tow** [comment]
Still working on https://github.com/enso-org/enso/pull/7832 — following the changes by Paweł in #7813 significant adjustments were needed — but the result will be much cleaner.

Other than that, some minor CI work.

**Comment by kazcw** [comment]
Finishing #7881: The generated code is working; the PR will be ready when I finish scripting the build.
